utils/resources_manager.py

```diff
@@ -28,8 +28,9 @@ def parse_index(self, index: str):
     def sync_fetch(self, filepath: str):
         """Fetches files in sync mode."""
         self.print_msg(f"Fetching {filepath}", "sync")
-        with requests.get(self.url + filepath) as resp:
-            return resp.content
+        resp = httpx.get(self.url + filepath)
```

Comment:
I think this change leads to connections never being closed, this should be verified